src/Microsoft.TemplateSearch.Common/BlobTemplateInfo.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Utils;
+using Newtonsoft.Json.Linq;
+
+#nullable enable
+
+namespace Microsoft.TemplateSearch.Common
+{
+    internal class BlobTemplateInfo : ITemplateInfo, IShortNameList
+    {
+        public BlobTemplateInfo(JObject jObject)
+        {
+            Identity = jObject.Value<string>(nameof(Identity));
+            Name = jObject.Value<string>(nameof(Name));
+            ShortNameList = jObject.Value<JArray>("ShortNameList").ToObject<string[]>();
+            Author = jObject.Value<string>(nameof(Author));
+            GroupIdentity = jObject.Value<string>(nameof(GroupIdentity));
+            //TODO: Check if this is actually needed for --search, or could it be just hard coded 0
```

Comment:
Afaiu, precedence is needed when choosing the template we want to instantiate. It looks like we don't need it for blob search. I'm OK with the current implementation or hardcoding it is zero. In either case, I suggest we remove this comment in this PR.